smmregrid/regrid.py

```diff
@@ -175,6 +175,10 @@ def apply_weights(source_data, weights, weights_matrix=None, masked=True, space_
 
         # apply the mask
         target_dask = dask.array.where(target_mask != 0.0, target_dask, numpy.nan)
+    
+    # after the tensordot, bring the NaN back in
+    # Use greatear/equal to avoid numerical noise from interpolation.
```

Comment:
typo: greater

Reply:
More seriously: how standard is it for NaN to 1e20 ... could it be something else? It depens on which value was set as NaN in the netcdf in theory. Would it be possible to parse it ?

Reply:
no this is done a few lines above before the tensordot but smmregrid itself! So it is totally safe.

Reply:
That would be _FillValue/missing_value

Reply:
what I am not 100% sure is if we get value slightly smaller thant 1e20 following interpolation. this never happened in my test but could be safer to 1e19?

Reply:
Ok, I just saw that line too. Ok. You are right with 1e19 .... well it could not harm to use that. Not very clean though. In theory 1e20 should be used as fillvalue, basically Nan, so it should not be really interpolated....but who knows

Reply:
New idea: instead of using a where, could you not do `ma.set_fill_value`on the result setting again fill value to 1e20 ? Would that not transform all 1e20 to NaN ?

Reply:
I think that the tensordot is done on the full matrix, so all the values are computed, also the 1e20. And this might create a bit of noise. I will set 1e19.

Reply:
Correction: "New idea: instead of using a where, could you not do ma.set_fill_valueon the result setting again fill value to 1e20 ? Would that not transform all 1e20 to NaN ?" 
 ma.set_fill_value seems to set the general value used for fill_value. So if dask does not see the result as NaN there is nothing to be done. Ok, I agree with the 1e19. Go with the fix

Reply:
In any case there should never be any NaN values, unless the input is masked or the entire frame is missing (they are all NaN). In this case this fix will work nicely anyway.